### PR TITLE
HOCS-1756

### PR DESCRIPTION
**Pull request by hubert-raich**
- added wiring for MPAM campaigns
- type-ahead widget now supports flat key-value input (default format
for entity list values)
- defaultOptions property added to type-ahead widget, when true
all options are visible on blank entry

### CODE REVIEW

File: src/shared/common/forms/type-ahead.jsx [hunk 1]
```diff
@@ -146,13 +160,15 @@ TypeAhead.propTypes = {
     label: PropTypes.string,
     name: PropTypes.string.isRequired,
     updateState: PropTypes.func.isRequired,
-    value: PropTypes.string
+    value: PropTypes.string,
+    defaultOptions: PropTypes.bool,
```
Comment (zuzak) [inline src/shared/common/forms/type-ahead.jsx:164]: hanging comma

File: src/shared/common/forms/type-ahead.jsx [hunk 2]
```diff
@@ -83,6 +96,7 @@ class TypeAhead extends Component {
                     error={error}
                     onChange={this.handleChange.bind(this)}
                     loadOptions={this.getOptions.bind(this)}
+                    defaultOptions={defaultOptions}
```
Comment (zuzak) [inline src/shared/common/forms/type-ahead.jsx:99]: is this okay to be `null`? (probably)

Comment (hubert-raich) [reply src/shared/common/forms/type-ahead.jsx:99]: Yes, in fact only one widget has it defined. Both undefined and false would mean the same thing in this instance.

File: src/shared/common/forms/type-ahead.jsx (hunk 1 above)
Comment (hubert-raich) [reply src/shared/common/forms/type-ahead.jsx:164]: fix pending

**Pull request merged**